packages/core/src/oidc/adapter.ts

```diff
@@ -38,13 +40,37 @@ export default function postgresAdapter(modelName: string): ReturnType<AdapterFa
   }
 
   return {
-    upsert: async (id, payload, expiresIn) =>
-      upsertInstance({
+    upsert: async (id, payload, expiresIn) => {
+      if (modelName === 'AccessToken' && payload.accountId) {
+        const application = payload.clientId ? await findApplicationById(payload.clientId) : null;
+
+        await insertUserLog({
```

Comment:
in general the "insert log" behavior shouldn't block the normal process, will it be better to put this in another middleware to proceed response regardless of the result of "insert log"? cc @simeng-li

Reply:
Had similar concerns.  Especially put

```js
 if (modalName === 'AccessToken' && payload.accounty) {
   // some log
 }
 ```
 
 into the business, logic feels a little bit hacky. 
 
 Sorry I was thinking of looking into it last week, but need to deep dive into the OIDCProvider adapter usage first.  Somehow put it aside for later.  @logto-io/eng feel free to share any insights.

Reply:
Follow Up:

Have couple of other questions:
1. Like @gao-sun  mentioned above ould extract the whole logging logic into a separate method or middleware handles the logging logic independent of the main adapter's `upsertInstance` stack. 
2. Just curious, in terms of the user log, besides Access Token data, do we need to log the grant update and session update as well? If so we might cover multiple scenarios not only just AccessToken Modal. Should we have a more generation way of implementing it?

Reply:
Let's put it aside, maybe wait for `audit` feature is set.